docs/plans/6.0-multi-vendor-marketplace.md

```
@@ -0,0 +1,630 @@
+# Targeting rule — grouped by subject_type: AND across dimensions, OR within one (Decision 4)
+class Spree::CommissionRule < Spree.base_class
+  belongs_to :commission_rate, class_name: 'Spree::CommissionRate'
+  belongs_to :subject, polymorphic: true, optional: true   # Spree::Product | Spree::Taxon | Spree::Vendor; nil ⇒ global
+end
+```
+
+**Resolution** (`Spree::Commissions::ResolveRate` service): load enabled `CommissionRate`s for the store ordered by `priority DESC`, filter by currency, return the first whose rules match the line item — rules grouped by `subject_type`, every present group needs ≥1 match (Decision 4). A rate with no rules is the global default. Seed default rates so `product.platform_fee || vendor.platform_fee` behavior is preserved without configuration — including the legacy store-preference default (30%) as the seeded global rate. This also untangles the three legacy meanings of `platform_fee` (store preference / vendor column / product override) into one model.
+
+**Calculation** (`Spree::Commissions::CalculateLine` service) — port `calculate_item_fee.rb`, generalized:
+
+```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)